src/main/java/jenkins/branch/WorkspaceLocatorImpl.java

```diff
@@ -438,6 +460,32 @@ private static synchronized void taskFinished() {
             Deleter.class.notifyAll();
         }
 
+        private static class CleanupTaskProvisioner implements Runnable{
+
+            @NonNull
+            private final TopLevelItem tli;
+
+            @NonNull
+            private final Queue<Node> nodes;
+
+            public CleanupTaskProvisioner(TopLevelItem tli, List<Node> nodes) {
+                this.tli = tli;
+                this.nodes = new LinkedList<>(nodes);
+            }
+
+            @Override
+            public void run() {
+                try {
+                    while (!nodes.isEmpty()){
+                        Deleter.acquireThread();
+                        Computer.threadPoolForRemoting.submit(new CleanupTask(tli, nodes.remove()));
+                    }
+                } catch (Exception e) {
```

Comment:
NIT:  if the exception is either a [`RejectedExecutionException`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/RejectedExecutionException.html) or an `InteruptedException` should we continue to process any of the nodes?

a rejectedExecution would seem to imply the execturor service was shut down so all future attempts to submit something would fail.
Likewise an `InteruptedException` somone wanted to interupt us whilst we where waiting to aquire a lock - normally that would be "please stop what you are doing gracefully"

Reply:
Hey @jtnord, I took a quick peek at this. As it is, wouldn't any exception terminate the loop? Or do you wish to only terminate the loop on either the `RejectedExecutionException`/`InterruptedException` and continue when encountering others?

Reply:
Hi @praqma-thi

Sorry about that when first reviewing it looked like I thought the `catch` was inside the while.